### DISCUSSION TITLE
float parsing from rgb(), rgba().

### DISCUSSION THREAD

**Discussion by rendarz** [mazznoer/csscolorparser-rs#11] [General]
I was thinking to add a feature for the `rgb()`, `rgba()` format: float parsing.
For instance: : `rgba(1.0f 0.0f 0.0f 1.0f)`
That's perfectly compatible with current format, and pretty useful in some circumstances.
I was thinking to add this by modifying the function `parse_percent_or_255` to `parse_percent_or_255_or_float`.
If you're ok with this, I will proceed to code it.
Cheers!

**Comment by mazznoer** [comment]
From https://developer.mozilla.org/en-US/docs/Web/CSS/color_value/rgb#values

> rgb(R, G, B[, A])
R (red), G (green), and B (blue) can be either [\<number\>](https://developer.mozilla.org/en-US/docs/Web/CSS/number)s or [\<percentage\>](https://developer.mozilla.org/en-US/docs/Web/CSS/percentage)s, where the number 255 corresponds to 100%.

And from https://developer.mozilla.org/en-US/docs/Web/CSS/number

> The \<number\> [CSS](https://developer.mozilla.org/en-US/docs/Web/CSS) [data type](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Types) represents a number, being either an integer or a number with a fractional component.

But number in the range [0, 255] can have a fractional component to. So how do we know if a number is in the range [0, 255] or in the range [0, 1] ?

**Reply by rendarz** [reply]
Simple, with 'f' after the number, they are in [0,1] range, otherwise they are in [0,255] range.
For instance: rgb(0.5, 0.5, 0.5) [0,255] range, while rgb(0.5f 0.5f 0.5f) in [0,1] range.

But this solution won't be compatible with CSS standard you mentioned. So maybe it's better to avoid that.
